### ISSUE TITLE
[JENKINS-3891] Build 'appears to be stuck' even if in queue

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#7530]
<p>If I understand the abort stuck builds feature, it is intended to kill builds<br/>
that hang due to some sort of execution.  However, builds are reported to be<br/>
stuck even if they are simply waiting their turn in the queue.  It seems like<br/>
this timer should not start until the job begins execution.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/pamdirac.jpg" title="pamdirac's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=pamdirac">pamdirac</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891" target="_blank">Build 'appears to be stuck' even if in queue</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke">kohsuke</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2009-06-23T15:36:37+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>If I understand the abort stuck builds feature, it is intended to kill builds
that hang due to some sort of execution.  However, builds are reported to be
stuck even if they are simply waiting their turn in the queue.  It seems like
this timer should not start until the job begins execution.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kohsuke.png" title="kohsuke's avatar" /> kohsuke:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891?focusedId=131123&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131123">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The intention of marking such jobs in the queue is to let you know that these
are starving for additional executors &#8212; it's time to add more slaves.</p>

<p>Perhaps we should change the wording a bit? Like from "stuck" to "starving
executors" or something like that? With perhaps a hyperlink to a page in Wiki
that talks about what you are supposed to do when you see it?</p>

<p>Or maybe we should change the color from red to something less scary, like yellow?</p>
</pre>
</details>

</ul>
</details>

<p>The intention of marking such jobs in the queue is to let you know that these<br/>
are starving for additional executors &#8212; it's time to add more slaves.</p>

<p>Perhaps we should change the wording a bit? Like from "stuck" to "starving<br/>
executors" or something like that? With perhaps a hyperlink to a page in Wiki<br/>
that talks about what you are supposed to do when you see it?</p>

<p>Or maybe we should change the color from red to something less scary, like yellow?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/pamdirac.jpg" title="pamdirac's avatar" /> pamdirac:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891?focusedId=131125&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131125">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ah, so I think I misunderstood the feature.  When I set a value for "Abort the
build if it's stuck", this same timeout applies separately to executor
starvation and the actual build execution time?</p>

<p>I guess what's weird is that we have different jobs that take wildly varying
lengths of time.  So I have timeouts for various jobs at 5 minutes, 30 minutes,
1 hour and 3 hours.  For a couple of hours during the day for us, it's very easy
to get a 15-minute queue depth.  The five minute jobs will then be reported as
stuck while sitting in the queue, but the other jobs won't.</p>

<p>I think that just a better indication of what is really happening is an ok solution.</p>
</pre>
</details>

</ul>
</details>

<p>Ah, so I think I misunderstood the feature.  When I set a value for "Abort the<br/>
build if it's stuck", this same timeout applies separately to executor<br/>
starvation and the actual build execution time?</p>

<p>I guess what's weird is that we have different jobs that take wildly varying<br/>
lengths of time.  So I have timeouts for various jobs at 5 minutes, 30 minutes,<br/>
1 hour and 3 hours.  For a couple of hours during the day for us, it's very easy<br/>
to get a 15-minute queue depth.  The five minute jobs will then be reported as<br/>
stuck while sitting in the queue, but the other jobs won't.</p>

<p>I think that just a better indication of what is really happening is an ok solution.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891?focusedId=131126&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131126">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : jglick
Path:
 trunk/hudson/main/core/src/main/resources/lib/hudson/queue.jelly
 trunk/hudson/main/war/images/hourglass.svg
 trunk/hudson/main/war/resources/images/16x16/hourglass.gif
 trunk/hudson/main/war/resources/images/24x24/hourglass.gif
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19163" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19163</a>
Log:
<span class="error">&#91;FIXED JENKINS-3891&#93;</span> Replaced misleading, verbose, and scary red "(appears to be stuck)" label with a warning icon &amp; tooltip.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : jglick<br/>
Path:<br/>
 trunk/hudson/main/core/src/main/resources/lib/hudson/queue.jelly<br/>
 trunk/hudson/main/war/images/hourglass.svg<br/>
 trunk/hudson/main/war/resources/images/16x16/hourglass.gif<br/>
 trunk/hudson/main/war/resources/images/24x24/hourglass.gif<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19163" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19163</a><br/>
Log:<br/>
<span class="error">&#91;FIXED JENKINS-3891&#93;</span> Replaced misleading, verbose, and scary red "(appears to be stuck)" label with a warning icon & tooltip.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891?focusedId=131127&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131127">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in hudson
User: : jglick
Path:
 trunk/www/changelog.html
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19164" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19164</a>
Log:
<a href="https://issues.jenkins.io/browse/JENKINS-3891" title="Build 'appears to be stuck' even if in queue" class="issue-link" data-issue-key="JENKINS-3891"><del>JENKINS-3891</del></a> Noting.</p>
</pre>
</details>

</ul>
</details>

<p>Code changed in hudson<br/>
User: : jglick<br/>
Path:<br/>
 trunk/www/changelog.html<br/>
<a href="http://fisheye4.cenqua.com/changelog/hudson/?cs=19164" class="external-link" target="_blank" rel="nofollow noopener">http://fisheye4.cenqua.com/changelog/hudson/?cs=19164</a><br/>
Log:<br/>
<a href="https://issues.jenkins.io/browse/JENKINS-3891" title="Build 'appears to be stuck' even if in queue" class="issue-link" data-issue-key="JENKINS-3891"><del>JENKINS-3891</del></a> Noting.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3891?focusedId=131128&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-131128">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3040 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>Issue 3040 has been marked as a duplicate of this issue. ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-23/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-3040%22">JENKINS-3040</a>]</i>
